app/src/main/java/com/smithnoff/mytaskyapp/domain/repository/UserRepository.kt

```diff
@@ -0,0 +1,6 @@
+package com.smithnoff.mytaskyapp.domain.repository
+
+interface UserRepository {
```

Comment:
I would rename this to something like
AuthenticationRepository

Since technically it doesn't have to do much with the user itself